### PR TITLE
Sidebar details

### PR DESCRIPTION
**Pull request by Machado21**
The sidebar details is working, when clicked in one NF-e open a sidebar with more information about the NF-e.

### CODE REVIEW

File: src/app/features/products-tracking/products-tracking.component.ts [hunk 1]
```diff
@@ -16,23 +17,31 @@ export class ProductsTrackingComponent implements OnInit {
...
-  getProduct(id: string) {
+  getProduct(id: ObjectId) {
     this.productService.getProductById(id).subscribe((product: Product) => {
       this.product = product;
     });
   }
+
+  callSidebar() {
+    this.details = 'sidebar_details';
+    console.log(this.details);
```
Comment (brunorcx) [inline src/app/features/products-tracking/products-tracking.component.ts]: Delete this file
```suggestion

```

File: src/app/features/tracking/tracking.component.less [hunk 2]
```diff
@@ -88,3 +88,56 @@
...
+  background-color: #fff;
+
+  transition: 0.5s;
+}
+.sidebar {
+  display: block;
+  position: fixed;
+
+  top: 0;
+  right: -51vw;
```
Comment (brunorcx) [inline src/app/features/tracking/tracking.component.less:113]: Consider changing this to use flex or grid

File: src/app/features/tracking/tracking.component.ts [hunk 3]
```diff
@@ -37,9 +48,22 @@ export class TrackingComponent implements OnInit {
...
       this.product = product;
     });
   }
+
+  openSidebar(logis: Logistic) {
+    this.showDetails = true;
+    this.details = 'sidebar_details';
+    this.backgrond = 'show_backgrond';
+
+    const data = logis; // Dados a serem passados
```
Comment (brunorcx) [inline src/app/features/tracking/tracking.component.ts]: Change comment to english
```suggestion
    const data = logis; // Data to be sent
```

File: src/app/shared/details/details.component.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        });
+    }
+  }
+
+  // getProducts() {
+  //   this.productService.getAllProducts().subscribe((products: Product[]) => {
+  //     this.products = products;
+  //   });
+  // }
+}
```
Comment (brunorcx) [inline src/app/shared/details/details.component.ts:48]: Delete this
```suggestion
```

**Review by brunorcx — CHANGES_REQUESTED** [review]

File: src/app/shared/details/details.component.ts (hunk 4 above)
Comment (Machado21) [inline src/app/shared/details/details.component.ts]: Can you verify the "}", it was deleted but wasn't comment.
on the line 48.

File: .vscode/launch.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+    "configurations": [
+        {
+            "type": "msedge",
+            "name": "http://localhost:4200/productsTracking",
+            "request": "launch",
+            "url": "http://localhost:4200/productsTracking"
+        }
+    ]
+}
```
Comment (Machado21) [inline .vscode/launch.json:10]:
```suggestion
}
```

File: src/app/shared/details/details.component.ts (hunk 4 above)
Comment (Machado21) [inline src/app/shared/details/details.component.ts:42]:
```suggestion
}
```

File: src/app/features/tracking/tracking.component.less (hunk 2 above)
Comment (Machado21) [reply src/app/features/tracking/tracking.component.less:113]: consider resolve this in the next commits

File: src/app/shared/details/details.component.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  getProduct(nfe: any) {
+    for (let i = 0; i < nfe.merchandise.length; i++) {
+      this.productService
+        .getProductById(nfe.merchandise[i])
+        .subscribe((product: Product) => {
+          this.productsNfe.push(product);
+        });
+    }
+  }
+
```
Comment (Machado21) [inline src/app/shared/details/details.component.ts]:
```suggestion
}
```

**Review by Machado21 — COMMENTED** [review]
Ok. Consider some things to fix in the next commits, and there is a "}" that was deleted but wasn't a comment.

**Comment by Machado21** [comment]
I accept some changes and extend one.

**Pull request merged**